ompi/mca/pml/ob1/pml_ob1_recvreq.h

```diff
@@ -178,7 +178,7 @@ recv_request_pml_complete(mca_pml_ob1_recv_request_t *recvreq)
 
         if(true == recvreq->req_recv.req_base.req_free_called) {
             if( MPI_SUCCESS != recvreq->req_recv.req_base.req_ompi.req_status.MPI_ERROR ) {
-                ompi_mpi_abort(&ompi_mpi_comm_world.comm, MPI_ERR_REQUEST);
+                OMPI_ERRHANDLER_INVOKE(OMPI_COMM_ERRORS, MPI_ERR_REQUEST, "Recv error after request freed.");
```

Comment:
Ideally, we should never abort internally, and should always pass the error to an error handler, eventually letting the end-user decide if the application should abort on error, or not. (see the following textual change to the standard in MPI-4: https://github.com/mpi-forum/mpi-standard/pull/9 ).

This PR made this change to avoid aborting in places where it is presumably not going to result in just crashing disorderly at a later point. The reason why this behavior has not been implemented consistently is that in some places it is uncertain/presumed to cause disorderly crashes if we do not force an immediate abort. Said changes are 'high quality', and can be postponed to a later PR without breaking strict compliance.

 Note that for this particular place (and its symmetric case for sendreqs), we are mandated to call the error handler on SELF, and avoid aborting if said handler is not an aborting kind; this is not expected to cause disorderly behavior. This is however the only place in the code where there is no place to 'bubble up' the error for an error handler invocation in the callers' context, as there will be no completion call on that request (other places exist where we do not bubble up errors, but there is a credible path to raise these errors in an appropriate API call, so that can be fixed over time).

Reply:
SELF or OMPI_COMM_ERRORS ?

The current error handling in MPI is finally streamlined: when there is a valid user facing MPI object, the error handler is called on it, and in all other cases we fall back to a default communicator (always a communicator). As such, I think I would have replaced `OMPI_ERRHANDLER_INVOKE(OMPI_COMM_ERRORS, ARGS...)` with a simpler macro `OMPI_COMM_DEFAULT_ERRHANDLER_INVOKE(ARGS)` and move all the logic of selecting the communicator (OMPI_COMM_ERRORS) in there.

Reply:
The new revision of this PR sends all 'unknown'  calls to `errhandler_invoke(NULL, NULL, -1...)` and let the function do the routing internally, there is no OMPI_COMM_ERRORS anymore).